### PR TITLE
183046852 service mutations

### PR DESCRIPTION
**Pull request by ZachCMP**
https://www.pivotaltracker.com/story/show/183046852

### CODE REVIEW

File: drivers/hmis/app/models/hmis/hud/validators/service_validator.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  end
+
+  def validate_type_provided(record)
+    return if TYPE_PROVIDED_MAP.any? { |rt, tp_map| record.record_type == rt && tp_map.keys.include?(record.type_provided) }
+
+    record.errors.add :type_provided, :invalid, message: 'Invalid TypeProvided for RecordType', full_message: "Value for TypeProvided '#{record.type_provided}' does not match RecordType '#{record.record_type}'"
+  end
+
+  def validate_sub_type_provided(record)
+    return if record.sub_type_provided.nil? && record.record_type != 144
```
Comment (eanders) [inline drivers/hmis/app/models/hmis/hud/validators/service_validator.rb]: is this the same as?
```suggestion
    return unless record.record_type == 144 && record.sub_type_provided.present?
```
If so, that seems more readable to me.

File: drivers/hmis/spec/requests/hmis/delete_service_spec.rb [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            type
+            options
+            __typename
+          }
+        }
+      }
+    GRAPHQL
+  end
+
+  it 'should create a service successfully' do
```
Comment (eanders) [inline drivers/hmis/spec/requests/hmis/delete_service_spec.rb]:
```suggestion
  it 'should delete a service successfully' do
```

File: drivers/hmis/spec/requests/hmis/update_service_spec.rb [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            type
+            options
+            __typename
+          }
+        }
+      }
+    GRAPHQL
+  end
+
+  it 'should create a service successfully' do
```
Comment (eanders) [inline drivers/hmis/spec/requests/hmis/update_service_spec.rb]:
```suggestion
  it 'should update a service successfully' do
```

**Review by eanders — APPROVED** [review]
Assuming tests pass, this looks fine, just a few minor comments.  I defer to @gigxz.

File: drivers/hmis/app/graphql/mutations/update_service.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
+module Mutations
+  class UpdateService < BaseMutation
```
Comment (gigxz) [inline drivers/hmis/app/graphql/mutations/update_service.rb:2]: It doesn't look like `dateUpdated` is changing on the update mutation

File: drivers/hmis/app/graphql/schema.graphql [hunk 5]
```diff
@@ -1236,6 +1300,26 @@ type Service {
   typeProvided: ServiceTypeProvided!
 }
 
+"""
+HUD Service Input
+"""
+input ServiceInput {
+  FAAmount: Float
+  clientId: ID
```
Comment (gigxz) [inline drivers/hmis/app/graphql/schema.graphql]: There's no need to require clientId on the input, we can get that from the enrollment. Lets remove this.

**Comment by ZachCMP** [comment]
@eanders All your changes looked good, so I committed them
@gigxz I fixed the two points of your feedback

Let me know if you need more work from me on this

**Review by gigxz — APPROVED** [review]
looks good, thanks!

**Pull request merged**